### PR TITLE
decision tree & upcaling

### PR DESCRIPTION
**Pull request by fvanderbiest**

### CODE REVIEW

File: thinkhazard_processing/processing.py [hunk 1]
```diff
@@ -48,6 +51,87 @@ def process(hazardset_id=None, force=False):
         process_hazardset(hazardset, force=force)
 
 
+def upscale_hazardcategories(target_adminlevel_mnemonic):
+    # identify the admin divisions matching the given level:
+    admindivs = DBSession.query(AdministrativeDivision)\
+        .join(AdminLevelType, AdministrativeDivision.leveltype_id == AdminLevelType.id) \
```
Comment (pgiraud) [inline thinkhazard_processing/processing.py]: You shouldn't have to specify on which the join is to be done. SQLAlchemy will determine it.

File: thinkhazard_processing/processing.py [hunk 2]
```diff
@@ -48,6 +51,87 @@ def process(hazardset_id=None, force=False):
...
+        .filter(AdminLevelType.mnemonic == target_adminlevel_mnemonic)
+    hazardtypes = DBSession.query(HazardType)
+    # for each of these admin divisions and hazard types,
+    # identify the highest HazardCategory among their children:
+    for admindiv in admindivs:
+        # identify max(level) for each hazardtype across children
+        for hazardtype in hazardtypes:
+            hazardcategory = DBSession.query(HazardCategory) \
+                .join((AdministrativeDivision, HazardCategory.administrativedivisions)) \
+                .join(HazardType, HazardCategory.hazardtype_id == HazardType.id) \
```
Comment (pgiraud) [inline thinkhazard_processing/processing.py]: No need to specify the relationship.

File: Makefile [hunk 3]
```diff
@@ -26,6 +27,10 @@ initdb: .build/requirements.timestamp
 process: .build/requirements.timestamp
 	.build/venv/bin/process
 
+.PHONY: decisiontree
+decisiontree: .build/requirements.timestamp
+	.build/venv/bin/decision_tree
```
Comment (arnaud-morvan) [inline Makefile:32]: I think it would be better to keep same names in scripts and makefile

File: thinkhazard_processing/tests/common.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+from thinkhazard_common.models import DBSession
+from ..models import Layer
+from sqlalchemy import func
+
+
+def new_geonode_id():
+    row = DBSession.query(func.max(Layer.geonode_id)).one_or_none()
+    if row[0] is None:
+        return 1
+    return row[0] + 1
```
Comment (pgiraud) [inline thinkhazard_processing/tests/common.py:10]: I think that @arnaud-morvan has added something similar to this already. To be checked.

Comment (fvanderbiest) [inline thinkhazard_processing/tests/common.py:10]: hence common

Comment (pgiraud) [inline thinkhazard_processing/tests/common.py:10]: This means that @arnaud-morvan code will have to be changed.

**Comment by pgiraud** [comment]
Please merge.

**Pull request merged**